### PR TITLE
Hoc loading

### PR DESCRIPTION
**Pull request by alejobh**
## Summary
Added a HOC for the components that have a loading state.
For using the hoc, we just wrap the component into `withLoading(Component, external)` where `external` is a flag that indicates if the Loading spinner goes to the center of the screen and its white, or if its a internal spinner that replaces an internal component like form actions.

### External Loading Spinner
image

### Internal Loading Spinner
image

#### Loading App and Getting profile information from API
image

#### Submitting a form
image

## trello cards
https://trello.com/c/mrxzotvb

### CODE REVIEW

File: react/game/src/app/components/withLoading/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+import styles from './styles.scss';
+
+const withLoading = (Component, external) => props => {
```
Comment (wfolini) [inline react/game/src/app/components/withLoading/index.js]: U can destructuring the props here

File: react/game/src/app/index.js [hunk 2]
```diff
@@ -3,14 +3,20 @@ import PropTypes from 'prop-types';
...
 
 class App extends Component {
   componentDidMount() {
     this.props.initApp();
   }
   render() {
     const { appLoading } = this.props;
-    return !appLoading ? <Router /> : <div>Loading</div>;
+    const external = true;
+    const WithLoading = withLoading(Router, external);
```
Comment (wfolini) [inline react/game/src/app/index.js]: U should use the HOC once, at the `export default ...`

**Review by wfolini — CHANGES_REQUESTED** [review]
👏  Great! Almost close !

File: react/game/src/app/components/withLoading/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  const { loading, msgLoading } = props;
+  const endClass = external ? 'External' : 'Internal';
```
Comment (jmtov) [inline react/game/src/app/components/withLoading/index.js]: I would change this name (External, Internal) for something a bit more descriptive. Maybe something that gives you an idea of what this variable will be used for without having to look at the whole code to understand it. 🙂

**Review by jmtov — APPROVED** [review]
I liked this new HOC! I made a comment. Please check it out!

File: react/game/src/app/index.js (hunk 2 above)
Comment (alejobh) [reply react/game/src/app/index.js]: ok ok 👌🏼

File: react/game/src/app/components/withLoading/index.js (hunk 1 above)
Comment (alejobh) [reply react/game/src/app/components/withLoading/index.js]: done! 😎

File: react/game/src/app/components/withLoading/index.js (hunk 3 above)
Comment (alejobh) [reply react/game/src/app/components/withLoading/index.js]: thanks for your comment kike, you are right!

**Review by wfolini — APPROVED** [review]
Almost done! 👏 👏

**Pull request merged**